cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1715,27 +1691,13 @@ parseMultiSigScript hms = do anyM <- any' hms
              case toBoundedInteger sci of
                Just reqInt ->
                  do msigscripts <- gatherMultiSigScripts listVectorValue
-                    Just <$> fromJSONmOfnChecks msigscripts reqInt (Vector.length listVectorValue)
+                    return . Just $ RequireMOf reqInt msigscripts
```

Comment:
Just trying to think if eliminating all checks on input is ok too...

This is what made me notice that `gatherMultiSigScripts` was using `catMaybes` which is dangerous.

I think perhaps we should keep one test here: the one that checks if the number of required successes is greater than the number of scripts in total, since that's almost certainly going to be a mistake.